src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -1790,5 +1790,538 @@ class B
+        const int {|Rename:V|} = 3;
+        return 1 + 2 + /*c3*/ V;
+    }
+}";
+
+            Test(code, expected, index: 2, compareTokens: false);
+        }
+
+        [WorkItem(528, "http://github.com/dotnet/roslyn/issues/528")]
+        [Fact(Skip = "http://github.com/dotnet/roslyn/issues/971"), Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void TestIntroduceLocalInExpressionBodiedMethodWithBlockBodiedAnonymousMethodExpression()
+        {
+            var code =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => delegate (int x)
+    {
+        return [|9|];
+    };
+}";
+
+            var expected =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => delegate (int x)
+    {
+        const int {|Rename:V|} = 9;
+        return V;
+    };
+}";
+
+            Test(code, expected, index: 2, compareTokens: false);
+        }
+
+        [WorkItem(528, "http://github.com/dotnet/roslyn/issues/528")]
+        [Fact(Skip = "http://github.com/dotnet/roslyn/issues/971"), Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void TestIntroduceLocalInExpressionBodiedMethodWithSingleLineBlockBodiedAnonymousMethodExpression()
+        {
+            var code =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => delegate (int x) { return [|9|]; };
+}";
+
+            var expected =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => delegate (int x) { const int {|Rename:V|} = 9; return V; };
+}";
+
+            Test(code, expected, index: 2, compareTokens: false);
+        }
+
+        [WorkItem(528, "http://github.com/dotnet/roslyn/issues/528")]
+        [Fact(Skip = "http://github.com/dotnet/roslyn/issues/971"), Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void TestIntroduceLocalInExpressionBodiedMethodWithBlockBodiedSimpleLambdaExpression()
+        {
+            var code =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => f =>
+    {
+        return f * [|9|];
+    };
+}";
+
+            var expected =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => f =>
+    {
+        const int {|Rename:V|} = 9;
+        return f * V;
+    };
+}";
+
+            Test(code, expected, index: 2, compareTokens: false);
+        }
+
+        [WorkItem(528, "http://github.com/dotnet/roslyn/issues/528")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void TestIntroduceLocalInExpressionBodiedMethodWithExpressionBodiedSimpleLambdaExpression()
+        {
+            var code =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y() => f => f * [|9|];
+}";
+
+            var expected =
+    @"using System;
+class TestClass
+{
+    Func<int, int> Y()
+    {
+        const int {|Rename:V|} = 9;
+        return f => f * V;
+    }
+}";
+
+            Test(code, expected, index: 2, compareTokens: false);
```

Comment:
Don't block checkin on this, but I think that the ideal in this case would be to change the lambda to not be a simple lambda instead of changing the property.  That way you avoid the capture.  On the other hand, it probably breaks expression trees. :frowning:.